v2/manager.go

```diff
@@ -568,32 +585,38 @@ func (c *Manager) MemoryEventFD() (int, uint32, error) {
 		return 0, 0, errors.New("failed to create inotify fd")
 	}
 	wd, err := syscall.InotifyAddWatch(fd, fpath, unix.IN_MODIFY)
-	if wd < 0 {
+	if err != nil {
 		syscall.Close(fd)
```

Comment:
I wonder if there is a way we could handle the possible errors coming from this `Close()` call.